api/content.go

```diff
@@ -0,0 +1,202 @@
+	// itemLink in alphabetical order
+	for _, href := range hrefs {
+		for _, item := range *itemLink {
+			if href == item.HRef {
+				var topicLink models.LinkObject
+				var selfLink models.LinkObject
+
+				selfLink.HRef = item.HRef
+				topicLink.ID = id
+				topicLink.HRef = "/topic/" + id
+
+				var cLinks models.ContentLinks
+				cLinks.Self = &selfLink
+				cLinks.Topic = &topicLink
+
+				var cItem models.ContentItem
+				cItem.Title = item.Title
+				cItem.Type = typeName
+				if privateResponse {
+					cItem.State = state
+				}
+				cItem.Links = &cLinks
+
+				if contentList.Items == nil {
+					contentList.Items = &[]models.ContentItem{cItem}
+				} else {
+					*contentList.Items = append(*contentList.Items, cItem)
+				}
+
+				count++
+			}
+		}
+	}
+
+	contentList.TotalCount = contentList.TotalCount + count
+}
+
+// getContentPublicHandler is a handler that gets content by its id from MongoDB for Web
+func (api *API) getContentPublicHandler(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	vars := mux.Vars(req)
+	id := vars["id"]
+	logdata := log.Data{
+		"request_id": ctx.Value(dprequest.RequestIdKey),
+		"content_id": id,
+		"function":   "getContentPublicHandler",
+	}
+
+	// check topic from mongoDB by id
+	_, err := api.dataStore.Backend.GetTopic(id)
+	if err != nil {
+		handleError(ctx, w, err, logdata)
+		return
+	}
+
+	// get content from mongoDB by id
+	content, err := api.dataStore.Backend.GetContent(id)
+	if err != nil {
+		// no content found
+		handleError(ctx, w, err, logdata)
+		return
+	}
+
+	// User is not authenticated and hence has only access to current sub document(s)
+
+	if content.Current == nil {
+		handleError(ctx, w, apierrors.ErrInternalServer, logdata)
+		return
+	}
+
+	var currentResult models.ContentResponseAPI
+
+	// Add spotlight first
+	addItem(&currentResult, "spotlight", content.Current.Spotlight, content.ID, content.Current.State, false)
+	// then Publications (alphabetically ordered)
+	addItem(&currentResult, "articles", content.Current.Articles, content.ID, content.Current.State, false)
+	addItem(&currentResult, "bulletins", content.Current.Bulletins, content.ID, content.Current.State, false)
+	addItem(&currentResult, "methodologies", content.Current.Methodologies, content.ID, content.Current.State, false)
+	addItem(&currentResult, "methodologyArticles", content.Current.MethodologyArticles, content.ID, content.Current.State, false)
+	// then Datasets (alphabetically ordered)
+	addItem(&currentResult, "staticDatasets", content.Current.StaticDatasets, content.ID, content.Current.State, false)
+	addItem(&currentResult, "timeseries", content.Current.Timeseries, content.ID, content.Current.State, false)
+
+	currentResult.Count = currentResult.TotalCount // This may be '0' which is the case for some existing ONS pages (like: bankruptcyinsolvency as of 3.feb.2021)
+
+	if err := WriteJSONBody(ctx, currentResult, w, logdata); err != nil {
+		return
```

Comment:
we probably want to at least log if there was an issue writing the response body

Reply:
WriteJSONBody log the error.